portal/models/i18n.py

```diff
@@ -12,27 +12,33 @@
 from subprocess import check_call
 from zipfile import ZipFile
 
-from .app_text import AppText
 from ..extensions import babel
+from .app_text import AppText
 from .intervention import Intervention
+from .organization import Organization
 from .user import current_user
 
 
 def get_db_strings():
-    elements = defaultdict(set)
-    for entry in AppText.query:
-        if entry.custom_text:
-            ct = re.sub('"', r'\\"', entry.custom_text)
-            elements['"{}"'.format(ct)].add("apptext: " + entry.name)
-    for entry in Intervention.query:
-        if entry.description:
-            desc = re.sub('"', r'\\"', entry.description)
-            elements['"{}"'.format(desc)].add("interventions: " + entry.name)
-        if entry.card_html:
-            ch = re.sub('"', r'\\"', entry.card_html)
-            elements['"{}"'.format(ch)].add("interventions: " + entry.name)
-    return elements
+    msgid_map = defaultdict(set)
+    i18n_fields = {
+        AppText: ('custom_text',),
+        Intervention: ('description', 'card_html'),
+        Organization: ('name',),
+    }
 
+    for model, fields in i18n_fields.items():
+        for entry in model.query:
+            for field_name in fields:
+                msgid = getattr(entry, field_name)
+                if not msgid:
+                    continue
+                msgid = '"{}"'.format(re.sub('"', r'\\"', msgid))
+                msgid_map[msgid].add("{model_name}: {field_ref}".format(
```

Comment:
I realize this is how it was done before, but why do we have a `set` of elements for the value side of this equation.  If you key on `msgid` how would you ever get to (or need) the second value in that set?

Reply:
The `msgid` is that actual string we translate, the set should only contain "references", ie some kind of reference to where it was extracted from, but are only kept for metadata/translation context. 

References are usually filenames and line numbers, but since these strings come from the DB we use what we can.